### ISSUE TITLE
[Bug] Tapping buttons with icons shows the URL of image instead

### ISSUE THREAD

**Issue by bogas04** [mozilla-mobile/fenix#17338]
[comment]: # (Please do your best to search for duplicate issues before filing a new issue so we can keep our issue board clean)
[comment]: # (Every issue should have the exact bug and steps to reproduce described in it. Please do not file feedback list tickets as it is difficult to parse them and address their individual points)
[comment]: # (Read https://github.com/mozilla-mobile/fenix#i-want-to-file-an-issue for more information)

## Steps to reproduce
1. Open amazon.in, kuvera.in or google.com
2. Tap on the hamburger or any other icon button (cross/menu etc)
3. Observe

### Expected behavior
The corresponding on click event handler should get executed always

### Actual behavior
The href or image src gets displayed in a dialog instead most of the times.
With Amazon, `javascript: void(0)`, while for other two sites the image src of the icon underneath the div/button gets displayed.

### Screencast

https://user-images.githubusercontent.com/6177621/103785146-dd9bf480-5060-11eb-9917-76f7cc486656.mp4

### Device information

* Android device: Realme X Android 10
* Fenix version: 85.0.0-beta.4 (Build #2015783155)
AC: 70.0.5, 70a0a5718
GV: 85.0-20201223151005
AS: 67.2.0

**Comment by kbrosnan** [comment]
Moved to bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1685344

Change performed by the [Move to Bugzilla add-on](https://addons.mozilla.org/en-US/firefox/addon/move-to-bugzilla/).

**Comment by kbrosnan** [comment]
Clicking/tapping is handled by Gecko

**Comment by bogas04** [comment]
Hi @kbrosnan, it's really nice of you to help me out here! 

In my testing it works fine only on Firefox Lite (I tried Firefox Stable, Beta, Nightly and Focus), which I means the issue is definitely with GeckoView and not Android-Components or such. Firefox Lite, IIRC uses Chromium.

I also observed that it works fine on a different phone (Moto G 5G), so it could be a problem only on specific devices or OS?

For reference, I've the issue on Realme X with RealmeUI 1.0 Android 10 (Qualcomm SDM710 Snapdragon 710 + Adreno 616) and not on Moto G 5G with Android 10 (Qualcomm SM7225 Snapdragon 750G 5G + Adreno 619).

This is such an annoying bug which basically breaks multiple websites with img buttons (hamburgers, add to cart +, etc) on it. I understand it might not be on all devices but surely it has rendered the browser frustrating to use for me.
